sql/analyzer/assign_update_join.go

```diff
@@ -55,25 +54,40 @@ func rowUpdatersByTable(ctx *sql.Context, node sql.Node, ij sql.Node) (map[strin
 	namesOfTableToBeUpdated := getTablesToBeUpdated(node)
 	resolvedTables := getTablesByName(ij)
 
-	ret := make(map[string]sql.RowUpdater)
+	rowUpdatersByTable := make(map[string]sql.RowUpdater)
+	for tableToBeUpdated, _ := range namesOfTableToBeUpdated {
+		resolvedTable, ok := resolvedTables[tableToBeUpdated]
+		if !ok {
+			return nil, plan.ErrUpdateForTableNotSupported.New(tableToBeUpdated)
+		}
 
-	for k, v := range resolvedTables {
-		if _, exists := namesOfTableToBeUpdated[k]; exists {
-			updatable, ok := v.Table.(sql.UpdatableTable)
-			if !ok {
-				continue
+		var updatable sql.UpdatableTable
+		switch tt := resolvedTable.Table.(type) {
+		case sql.UpdatableTable:
+			updatable = tt
+		case *plan.ProcessTable:
```

Comment:
`sql.TableWrapper.Underlying()` might simplify these last two cases

Reply:
something like
```go
	var table = rt.Table
	if t, ok := table.(sql.TableWrapper); ok {
		table = t.Underlying()
	}
	updatable, ok := table.(sql.Updateable)
```

Reply:
oooh, very nice. thanks for pointing out `TableWrapper`.